windows-apps-src/design/input/keyboard-events.md

```diff
@@ -103,13 +103,15 @@ The [**KeyDown**](https://docs.microsoft.com/uwp/api/windows.ui.xaml.uielement.k
 
 Modifier keys are keys such as Ctrl or Shift that users typically press in combination with other keys. Your app can use these combinations as keyboard shortcuts to invoke app commands.
 
-You detect shortcut key combinations by using code in your [**KeyDown**](https://docs.microsoft.com/uwp/api/windows.ui.xaml.uielement.keydown) and [**KeyUp**](https://docs.microsoft.com/uwp/api/windows.ui.xaml.uielement.keyup) event handlers. You can then track the pressed state of the modifier keys you are interested in. When a keyboard event occurs for a non-modifier key, you can check whether a modifier key is in the pressed state at the same time.
+You detect shortcut key combinations by using code in your [**KeyDown**](https://docs.microsoft.com/uwp/api/windows.ui.xaml.uielement.keydown) and [**KeyUp**](https://docs.microsoft.com/uwp/api/windows.ui.xaml.uielement.keyup) event handlers. Within those handlers, you can track the pressed state of the modifier keys you are interested in (by storing whether or not Ctrl is pressed, for example). When a keyboard event occurs for a non-modifier key, you can check whether a modifier key is in the pressed state at the same time.
+
+Alternatively, you can use the [**GetKeyState()**](https://docs.microsoft.com/uwp/api/windows.ui.core.corewindow.getkeystate) function of the [**CoreWindow**](https://docs.microsoft.com/uwp/api/windows.ui.core.corewindow) (available via [**CoreWindow::GetForCurrentThread()**](https://docs.microsoft.comuwp/api/windows.ui.core.corewindow.getforcurrentthread), for example) to check modifier states on-demand when a non-modifier key is pressed.
+
+The following examples implement this second method while also including stub code for the first implementation.
 
 > [!NOTE]
 > The Alt key is represented by the **VirtualKey.Menu** value.
 
- 
-
 ### Shortcut keys example
 
```

Comment:
The examples given don't fully implement the first strategy.

They include `KeyDown` handlers to store when ctrl is first pressed, but they don't ever update that value when the key is up:

```c++
void KeyboardSupport::MainPage::Grid_KeyDown(Platform::Object^ sender, Windows::UI::Xaml::Input::KeyRoutedEventArgs^ e)
{
    if (e->Key == VirtualKey::Control) isCtrlKeyPressed = true;
}


void KeyboardSupport::MainPage::Grid_KeyUp(Platform::Object^ sender, Windows::UI::Xaml::Input::KeyRoutedEventArgs^ e)
{
    if (IsCtrlKeyPressed()) 
    {
        if (e->Key==VirtualKey::P) { DemoMovie->Play(); }
        if (e->Key==VirtualKey::A) { DemoMovie->Pause(); }
        if (e->Key==VirtualKey::S) { DemoMovie->Stop(); }
    }
}
```

The C# example does not have this extra code (though I think the samples should include both methods and explain them).